ext/nnstreamer/tensor_filter/tensor_filter_llamacpp.cc

```diff
@@ -0,0 +1,339 @@
+{
+  public:
+  TensorFilterLlamaCpp ();
+  ~TensorFilterLlamaCpp ();
+
+  /* mandatory methods */
+  tensor_filter_subplugin &getEmptyInstance ();
+  void configure_instance (const GstTensorFilterProperties *prop);
+  void invoke (const GstTensorMemory *input, GstTensorMemory *output);
+  void invoke_dynamic (GstTensorFilterProperties *prop,
+      const GstTensorMemory *input, GstTensorMemory *output);
+  void getFrameworkInfo (GstTensorFilterFrameworkInfo &info);
+  int getModelInfo (model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info);
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+
+  /* static methods */
+  static void init_filter_llama ();
+  static void fini_filter_llama ();
+
+  private:
+  static TensorFilterLlamaCpp *registered;
+  static const char *name;
+  static const accl_hw hw_list[];
+  static const int num_hw;
+
+  llama_context *ctx;
+  llama_model *model = nullptr;
+  llama_sampler *smpl = nullptr;
+  llama_model_params model_params;
+  llama_sampler_chain_params sampler_params;
+  llama_context_params ctx_params;
+  llama_batch batch;
+};
+
+void init_filter_llama (void) __attribute__ ((constructor));
+void fini_filter_llama (void) __attribute__ ((destructor));
+
+TensorFilterLlamaCpp *TensorFilterLlamaCpp::registered = nullptr;
+const char *TensorFilterLlamaCpp::name = "llamacpp";
+
+/**
+ * @brief Construct a new llamacpp subplugin instance
+ */
+TensorFilterLlamaCpp::TensorFilterLlamaCpp ()
+{
+}
+
+/**
+ * @brief Destructor of TensorFilterllamacpp
+ */
+TensorFilterLlamaCpp::~TensorFilterLlamaCpp ()
+{
+}
+
+/**
+ * @brief Method to get an empty object
+ */
+tensor_filter_subplugin &
+TensorFilterLlamaCpp::getEmptyInstance ()
+{
+  return *(new TensorFilterLlamaCpp ());
+}
+
+/**
+ * @brief Configure llamacpp instance
+ */
+void
+TensorFilterLlamaCpp::configure_instance (const GstTensorFilterProperties *prop)
+{
+  /** @todo: Get ngl value from prop */
+  int ngl = 99;
+
+  /** load dynamic backends */
+  ggml_backend_load_all ();
+  model_params = llama_model_default_params ();
+  model_params.n_gpu_layers = ngl;
+
+  model = llama_load_model_from_file ((char *) prop->model_files[0], model_params);
+  if (model == nullptr) {
+    throw std::invalid_argument ("Failed to load model");
+  }
+
+  ctx_params = llama_context_default_params ();
+
+  /** enable performance counters */
+  ctx_params.no_perf = false;
+
+  sampler_params = llama_sampler_chain_default_params ();
+  sampler_params.no_perf = false;
+  smpl = llama_sampler_chain_init (sampler_params);
+
+  llama_sampler_chain_add (smpl, llama_sampler_init_greedy ());
+}
+
+/**
+ * @brief Invoke llamacpp using input tensors
+ */
+void
+TensorFilterLlamaCpp::invoke (const GstTensorMemory *input, GstTensorMemory *output)
+{
+  UNUSED (input);
+  UNUSED (output);
+
+  throw std::runtime_error (
+      "llamacpp only supports invoke-dynamic mode. Set `invoke-dynamic=true`");
```

Comment:
Please check and return error at the init phase (CAPS negotiation or configure_instance, whichever earlier possible)

Or...

Find a way to force-enable invoke-dynamic (change the property) at initialization phase and emit warning if it wasn't set true.

Reply:
`prop` value is const at the init phase and inaccessible from anywhere else so I added exception instead of change the property.
Thank you!